Student/StudentUnitTests/Submissions/SubmissionButton/SubmissionButtonPresenterTests.swift

```diff
@@ -0,0 +1,302 @@
+        XCTAssertNil(view.presented)
+        XCTAssert(router.calls.isEmpty)
+    }
+
+    func testPickFilesEmptyExtensions() {
+        let a = Assignment.make([ "submissionTypesRaw": [ "online_upload" ], "allowedExtensions": [] ])
+        presenter.pickFiles(for: a)
+        let filePicker = (view.presented as? UINavigationController)?.viewControllers.first as? FilePickerViewController
+        XCTAssertEqual(filePicker?.sources, [.files, .library, .camera])
+    }
+
+    func testPickFilesFilesOnly() {
+        let a = Assignment.make([ "submissionTypesRaw": [ "online_upload" ], "allowedExtensions": [ "txt" ] ])
+        presenter.pickFiles(for: a)
+        let filePicker = (view.presented as? UINavigationController)?.viewControllers.first as? FilePickerViewController
+        XCTAssertEqual(filePicker?.sources, [.files])
+    }
+
+    func testPickFilesImages() {
+        let a = Assignment.make([ "submissionTypesRaw": [ "online_upload" ], "allowedExtensions": [ "jpg" ] ])
+        presenter.pickFiles(for: a)
+        let filePicker = (view.presented as? UINavigationController)?.viewControllers.first as? FilePickerViewController
+        XCTAssertEqual(filePicker?.sources, [.files, .library, .camera])
+    }
+
+    func testAddFile() {
+        let url = URL(fileURLWithPath: "/file.txt")
+        presenter.add(filePicker, url: url)
+        XCTAssertEqual(presenter.files.count, 1)
+    }
+
+    func testSubmitFiles() {
+        let url = URL(fileURLWithPath: "/file.txt")
+        XCTAssertFalse(presenter.canSubmit(filePicker))
+        presenter.add(filePicker, url: url)
+        XCTAssertTrue(presenter.canSubmit(filePicker))
+        presenter.submit(filePicker)
+        XCTAssert(filePicker.dismissed)
+        XCTAssertEqual(fileUploader.uploads.count, 1)
+    }
+
+    func testRetryFileUpload() {
+        XCTAssertNoThrow(presenter.retry(filePicker))
+    }
+
+    func testCancelFileUpload() {
+        let url = URL(fileURLWithPath: "/file.txt")
+        presenter.add(filePicker, url: url)
+        XCTAssertEqual(presenter.files.count, 1)
+        presenter.cancel(filePicker)
+        XCTAssertEqual(presenter.files.count, 0)
+        XCTAssertEqual(fileUploader.cancels.count, 1)
+    }
+
+    func testPickMediaRecordingType() {
+        presenter.pickMediaRecordingType()
+        XCTAssert(view.presented is UIAlertController)
+    }
+
+    func testCancelAudioRecording() {
+        presenter.cancel(audioRecorder)
+        XCTAssert(audioRecorder.dismissed)
+    }
+
+    func testSendAudioRecording() {
+        let url = URL(string: "data:audio/x-aac,")!
+        presenter.send(audioRecorder, url: url)
+        XCTAssertNotNil(view.presented)
+    }
+
+    func testImagePickerControllerVideo() {
+        let url = URL(string: "data:audio/x-aac,")!
+        presenter.imagePickerController(imagePicker, didFinishPickingMediaWithInfo: [.mediaURL: url])
+        XCTAssert(imagePicker.dismissed)
+    }
+/*
+    func testSubmitMediaType() {
```

Comment:
If it's too difficult to unit test these, I'm fine with removing all of it. Media submissions is on the list for UI tests. We'll make sure to cover them well there.